.github/copilot-instructions.md

```diff
@@ -0,0 +1,262 @@
+# Copilot Instructions for Modular Avatar
+
+## Project Overview
+
+Modular Avatar is a suite of **non-destructive** tools for modularizing VRChat avatars and distributing avatar components. It allows users to easily add outfits, accessories, and gimmicks to avatars through drag-and-drop functionality.
+
+### Key Technologies
+- **Unity C#** - Core functionality for Unity Editor and runtime
+- **VRChat SDK3** - Integration with VRChat avatar systems  
+- **NDMF (Non-Destructive Modular Framework)** - Build system for non-destructive avatar modifications
+- **UI Toolkit** - Modern Unity UI system for editor interfaces
+- **Docusaurus 3.8.1** - Documentation website with internationalization
+- **Localization** - Multi-language support for both code and documentation
+
+### Repository Structure
+- `Runtime/` - Core runtime components and systems
+- `Editor/` - Unity Editor tools and inspectors
+- `docs~/` - Docusaurus documentation site
+  - `docs/` - English documentation
+  - `i18n/ja/docusaurus-plugin-content-docs/current/` - Japanese documentation
+- `UnitTests~/` - Unit tests for the project
+- `Samples/` - Example prefabs and usage samples
+
+## Code Style and Guidelines
+
+### C# Coding Standards
+
+#### General Principles
+- Follow Microsoft C# coding conventions
+- Use clear, descriptive names for variables, methods, and classes
+- Prefer composition over inheritance
+- Write self-documenting code with meaningful comments only when necessary
+- Use nullable reference types where appropriate
+
+#### Naming Conventions
+```csharp
+// Classes and interfaces: PascalCase
+public class MergeArmature : AvatarTagComponent
+public interface ILocalizationProvider
+
+// Methods and properties: PascalCase  
+public void ProcessAvatar()
+public string DisplayName { get; set; }
+
+// Private fields: camelCase with underscore prefix
+private GameObject _targetObject;
+private readonly Dictionary<string, float> _propertyOverrides;
+
+// Parameters and local variables: camelCase
+public void SetValue(string parameterName, float value)
+
+// Constants: PascalCase
+public const string DefaultPrefix = "MA_";
+```
+
+#### Unity-Specific Guidelines
+- Always null-check Unity objects before use (`if (obj != null)`)
+- Use `[SerializeField]` for private fields that need serialization
+- Implement proper `OnValidate()` methods for editor-time validation
+- Use `[HideInInspector]` judiciously to keep inspectors clean
+- Leverage Unity's component lifecycle methods appropriately
+
+#### Component Architecture
+- Most MA components inherit from `AvatarTagComponent`
+- Use `[DefaultExecutionOrder]` attributes to control component initialization order
+- Implement `OnValidate()` for editor-time validation and UI updates
```

Comment:
Remove. We do not want to damage saved data by changing it on load using `OnValidate`.